eudemo/eudemo/equatorial_port.py

```diff
@@ -0,0 +1,200 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021-2023 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh,
+#                         J. Morris, D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+EU-DEMO Equatorial Port
+"""
+from dataclasses import dataclass
+from numpy import ndarray, transpose
+from typing import Dict, Type, Union
+
+from bluemira.base.builder import Builder, ComponentManager
+from bluemira.base.components import Component, PhysicalComponent
+from bluemira.base.designer import Designer
+from bluemira.base.parameter_frame import Parameter, ParameterFrame
+from bluemira.display.palettes import BLUE_PALETTE
+from bluemira.geometry.face import BluemiraFace
+from bluemira.geometry.tools import (
+    boolean_fuse,
+    circular_pattern,
+    extrude_shape,
+    make_polygon,
+)
+from bluemira.geometry.wire import BluemiraWire
+
+
+class EquatorialPort(ComponentManager):
+    """
+    Wrapper around a Equatorial Port component tree
+    """
+
+    def xz_boundary(self) -> BluemiraWire:
+        """Returns a wire defining the x-z boundary of the Equatorial Port"""
+        return (
+            self.component.get_component("xz")
+            .get_component(EquatorialPortBuilder.KOZ)
+            .shape.boundary[0]
+        )
+
+
+@dataclass
+class EquatorialPortDesignerParams(ParameterFrame):
+    """
+    Equatorial Port Designer parameters
+    """
+
+    x_ib: Parameter[float]
+    x_ob: Parameter[float]
+    ep_height: Parameter[float]
+
+
+@dataclass
+class EquatorialPortBuilderParams(ParameterFrame):
+    """
+    Equatorial Port builder parameters
+    """
+
+    n_EP: Parameter[int]
+    ep_coordinates: Parameter[ndarray]
+    ep_r_corner: Parameter[float]
+
+
+class EquatorialPortDesigner(Designer):
+    """
+    Equatorial Port Designer
+    """
+
+    param_cls: Type[EquatorialPortDesignerParams] = EquatorialPortDesignerParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, EquatorialPortDesignerParams],
+        build_config: Union[Dict, None],
+    ):
+        super().__init__(params, build_config)
+

```

Comment:
Obviously this will need to be threaded through.
eg:  
```suggestion
        build_config: Union[Dict, None],
        x_ib: float,
        x_ob: float,
    ):
        super().__init__(params, build_config)
        self.x_ib = x_ib
        self.x_ob = x_ob


```